src/chain/chaininterface.rs

```diff
@@ -99,11 +132,31 @@ impl ChainWatchInterface for ChainWatchInterfaceUtil {
 		let mut vec = self.listeners.lock().unwrap();
 		vec.push(listener);
 	}
+
+	fn get_network(&self) -> Network {
+		self.network
+	}
+
+
+	fn get_chain_txo(&self, genesis_hash: Sha256dHash, _txid: Sha256dHash, _output_index: u16) -> Result<(Script, u64), ChainError> {
+		watched_chain!(self, genesis_hash);
+
+		//TODO: self.BlockchainStore.get_txo(txid, output_index)
```

Comment:
Add chain_store: Arc<ChainStore> in ChainWatchInterface? Need to have a look on rust-blockchain-store